### DISCUSSION TITLE
Suggested schema configuration for provided docker-compose stack?

### DISCUSSION THREAD

**Discussion by arpieb** [vesoft-inc/nebula#5798] [Q&A]
Hi there, first off thanks for releasing and supporting such a powerful graph DB as an OSS project!  I'm performing some research on a graph using what I realize are somewhat limited resources:

- 40TB striped ZFS volume (HDD)
- 128GB RAM
- 1x i7-7800X (6-core, 12-thread)
- Ubuntu Server 22.04 LTS

The graph in question is at the following scale:

- Total Vertices Count: 82,378,983
- Total Edges Count: 271,161,214

I am currently running it using the default 3-node docker compose stack, which seems to work fine for imports and simple traversal, but I start having issues with anything more than 3-4 hop path lengths (RPC timeouts, container crashes).  I noticed there are some tuning parameters related to schema definition, was wondering if the defaults are fine for the docker-compose stack or if I should redefine the schema with more reasonable tuning parameters for that setup?

This is definitely not a production machine, just a lab server, so prod-level fault-tolerance is not a requirement, just trying to squeeze the best performance out of what I have to work with.  If  there is a better single-machine configuration you could recommend for R&D work, I'm also all ears.

Thanks!

**Comment by QingZ11** [comment]
@wey-gu could you take a look at this issue?

**Comment by QingZ11** [comment]
> was wondering if the defaults are fine for the docker-compose stack or if I should redefine the schema with more reasonable tuning parameters for that setup?

Sorry for the late reply, I have consulted with the relevant development personnel, and those parameter options are also applicable to the Docker environment. Thank you for your interest and support in Nebula.
